server/remark-toc.ts

```diff
@@ -0,0 +1,120 @@
+import * as nodeFS from "fs";
+import path from "path";
+import matter from "gray-matter";
+import { visitParents } from "unist-util-visit-parents";
+import { fromMarkdown } from "mdast-util-from-markdown";
+import type { Parent } from "unist";
+import type { VFile } from "vfile";
+import type { Content } from "mdast";
+import type { Transformer } from "unified";
+
+// relativePathToFile takes a filepath and returns a path we can use in links
+// to the file in a table of contents page. The link path is a relative path
+// to the directory where we are placing the table of contents page.
+// @param root {string} - the directory path to the table of contents page.
+// @param filepath {string} - the path from which to generate a link path.
+const relativePathToFile = (root: string, filepath: string) => {
+  // Return the filepath without the first segment, removing the first
+  // slash. This is because the TOC file we are generating is located at
+  // root.
+  return filepath.slice(root.length).replace(/^\//, "");
+};
+
+// getTOC generates a list of links to all files in the same directory as
+// filePath except for filePath. The return value is an object with two
+// properties:
+// - result: a string containing the resulting list of links.
+// - error: an error message encountered during processing
+export const getTOC = (filePath: string, fs = nodeFS) => {
+  const dirPath = path.dirname(filePath);
+  if (!fs.existsSync(dirPath)) {
+    return {
+      error: `Cannot generate a table of contents for nonexistent directory at ${dirPath}`,
+    };
+  }
+
+  const { name } = path.parse(filePath);
+
+  const files = fs.readdirSync(dirPath, "utf8");
+  let mdxFiles = new Set();
+  const dirs = files.reduce((accum, current) => {
+    // Don't add a TOC entry for the current file.
+    if (name == path.parse(current).name) {
+      return accum;
+    }
+    const stats = fs.statSync(path.join(dirPath, current));
+    if (!stats.isDirectory() && current.endsWith(".mdx")) {
+      mdxFiles.add(path.join(dirPath, current));
+      return accum;
+    }
+    accum.add(path.join(dirPath, current));
+    return accum;
+  }, new Set());
+
+  // Add rows to the menu page for non-menu pages.
+  let entries = [];
```

Comment:
nit: i don't see this reference updated anywhere so we can change to `const`

Reply:
👍 Fixed in 74911f268e118f7310f45195a6b48bf1b6338094 (along with a `.remarkrc` update to stop this from breaking the remark linter).